src/main/scala/com.snowplowanalytics.iglu/ctl/Storage.scala

```diff
@@ -25,37 +25,107 @@ case class Storage[F[_]](xa: Transactor[F]) {
     * Fetch columns of the given database table
     */
   def getColumns(tableName: String, tableSchema: String)(implicit F: Bracket[F, Throwable]): F[List[Storage.Column]] = {
-    (fr"SELECT column_name FROM information_schema.columns" ++
-      fr"WHERE" ++
-      fr"table_name = $tableName" ++
-      fr"AND table_schema = $tableSchema" ++
-      fr"ORDER BY ordinal_position").query[Storage.Column].to[List].transact(xa)
+    queryInformationSchema(tableName, tableSchema)
+      .map {
+        _.map { row =>
+          val `type` = mapRowDataType(row)
+          Storage.Column(row.name, `type`, row.isNullable)
+        }
+      }
   }
 
-  def getComment(tableName: String, tableSchema: String)(implicit F: Bracket[F, Throwable]) =
-    fr"SELECT oid, obj_description(oid) AS comment FROM pg_class WHERE relname = $tableName"
-      .query[Storage.Comment].option.transact(xa)
+  def getComment(tableName: String)(implicit F: Bracket[F, Throwable]): F[Option[Storage.Comment]] =
+    fr"SELECT obj_description(oid) AS comment FROM pg_class WHERE relname = $tableName"
+      .query[Storage.CommentRow].option.transact(xa)
+      .map(_.flatMap(_.objDescription.map(Storage.Comment)))
+
+  private def queryInformationSchema(tableName: String, tableSchema: String)
+                                    (implicit F: Bracket[F, Throwable]): F[List[Storage.InformationSchemaRow]] = {
+    sql"""SELECT
+          column_name,
+          data_type,
+          is_nullable,
+          character_maximum_length,
+          numeric_precision,
+          numeric_scale 
+          FROM information_schema.columns
+          WHERE table_name = $tableName
+          AND table_schema = $tableSchema  
+          ORDER BY ordinal_position""".query[Storage.InformationSchemaRow].to[List].transact(xa)
+  }
+
+  private def mapRowDataType(row: Storage.InformationSchemaRow): Column.Type = {
+    row.dataType match {
+      case "timestamp without time zone" => Storage.Column.Type.Timestamp
+      case "data" => Storage.Column.Type.Date
+      case "integer" => Storage.Column.Type.Int
+      case "smallint" => Storage.Column.Type.SmallInt
+      case "bigint" => Storage.Column.Type.BigInt
+      case "double precision" => Storage.Column.Type.Double
+      case "numeric" => Storage.Column.Type.Decimal(row.precision, row.scale)
+      case "boolean" => Storage.Column.Type.Boolean
+      case "character varying" => Storage.Column.Type.Varchar(row.maxLength.getOrElse(256))
+      case "character" => Storage.Column.Type.Char(row.maxLength.getOrElse(1))
+      case other => throw new RuntimeException(s"Unknown storage data type - $other")
+    }
+  }
 }
 
 object Storage {
 
-  case class Column(columnName: String)
+  val commonColumns: List[Column] = List(
+    Column("schema_vendor", Column.Type.Varchar(128), isNullable = false),
```

Comment:
Schema-ddl has these definitions.

Reply:
>I found only 'stringified' version [here](https://github.com/snowplow/schema-ddl/blob/b1173cad7ae8cd211207dbd38115640a6cb0afc0/modules/core/src/main/scala/com.snowplowanalytics/iglu.s>chemaddl/redshift/ShredModelEntry.scala#L98) and I thought it would be nice to have 'typed' version defined, so that's what I did here. Maybe it's somewhere in schema-ddl and I just couldn't find it :P

It had been there but I removed it on later iterations. 

Those columns could just be filtered out by name as they don't require verification, as you suggested.